crossdock/server/server.py

```diff
@@ -130,79 +168,88 @@ def handle_trace_request(self, http_request, trace_request, span_handler, respon
         if span_handler:
             span_handler(span)
 
-        trace_id = "%x" % span.trace_id
-        observed_span = ObservedSpan(
-            trace_id, span.is_sampled(),
-            span.get_baggage_item(constants.baggage_key))
-
-        tr = TraceResponse(span=observed_span)
+        with tchannel.context_provider.span_in_context(span):
```

Comment:
while functionally the same, this is an HTTP server and it's better if it used `span_in_stack_context` method from `opentracing_instrumentation` (more testing this way)